kong/resty/dns/client.lua

```diff
@@ -839,7 +812,36 @@ local function syncQuery(qname, r_opts, try_list)
   local key = qname..":"..r_opts.qtype
   local item = queue[key]
 
-  -- if nothing is in progress, we start a new async query
+  local is_yieldable = in_yieldable_phase()
+
+  -- If nothing is in progress and in the yieldable phase,
+  -- we start a new sync query
+
+  if not item and is_yieldable then
```

Comment:
I'd like not to do this:

```lua
if not item and is_yieldable then
...
end

if not item then
...
end


```

instead, I prefer:

```lua
if not item then
  if is_yieldable then
  ...
  else
end
```

And with further refactoring, with a reversed condition, we could do this:

```lua
if item then
  ...
  return ...
end

if is_yieldable then
  ...
else
 ...
end
```

At least, let's not check the not item condition twice.

Reply:
resolved